<a href="https://colab.research.google.com/github/estefania-gherra/AnalisisyVisualizacion/blob/master/Tefi_Resoluci%C3%B3n_Ejercicio_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>


## Ejercicio

Responder:

**¿Si uno tiene más de 5 años de experiencia, la probabilidad de cobrar más que el promedio aumenta?**

**¿Estos eventos, son independientes?**




## Respuestas:

Para decir que "la probabilidad de cobrar más que el promedio aumenta si uno tiene más de cinco años de experiencia", hay que verificar:

      P (A|B) > P (A)


donde:

      A = cobrar más que el promedio (A_is_above_avg)
      B = contar con más de 5 años de experiencia (B_experience_greater_5)

Para calcular P(A|B) recordar que:

 $$P(A|B) = \frac{P(A \cap B)}{P(B)}$$



**Cálculos:**

In [ ]:
# 1- Importar librerías y cargar el dataset

import io
import matplotlib
import matplotlib.pyplot as plt
import numpy
import pandas as pd
import seaborn

seaborn.set_context('talk')

url = 'https://cs.famaf.unc.edu.ar/~mteruel/datasets/diplodatos/sysarmy_survey_2020_processed.csv'
df = pd.read_csv(url)
#df[:3]


In [ ]:
# 1.2- variable salario

salary_col = 'salary_monthly_NETO'

avg_salary = df[salary_col].mean()
#avg_salary

In [ ]:
# 1.3- variable experiencia

experience_col = 'profile_years_experience'


In [ ]:
# 2- Cálculo de probabilidades

# 2.1-    P(A):

A_is_above_avg = df[salary_col] > avg_salary
#print (A_is_above_avg)

prob_A = sum(A_is_above_avg)  /  len(df)
prob_A



0.33125512715340444

In [ ]:
# 2.2-    P(B):

B_experience_greater_5 = df[experience_col] > 5
#print (B_experience_greater_5)

prob_B = sum(B_experience_greater_5) / len(df)
prob_B

0.5873666940114848

In [ ]:
# 2.3-    P(A ∩ B):

A_intersec_B = df[A_is_above_avg  &  B_experience_greater_5]
#A_intersec_B

prob_A_intersec_B = len(A_intersec_B) / len(df)
prob_A_intersec_B


0.26808859721082856

In [ ]:
# 2.4- calcular P(A|B):

prob_A_dado_B = prob_A_intersec_B / prob_B
prob_A_dado_B

0.45642458100558664

# **Conclusión:**

Como

**P (A|B) = 0.456  > P (A) = 0.331**

entonces  

**SÍ AUMENTA la probabilidad de cobrar más que el promedio, al tener más de cinco años de experiencia. Y por lo tanto los eventos NO SON INDEPENDIENTES.**



Concluyo esto porque la independencia entre A y B se puede probar de dos formas:

1) Si P(A|B) = P(A)  entonces A y B son independientes.

Como vimos arriba que P(A|B) > P(A)  

**entonces  A y B NO SON INDEPENDIENTES. Hay una relación entre A y B.** 


2) Si P(A∩B) = P(A)*P(B)   entonces A y B son independientes.

0.268 ≠ 0.1945

entonces A y B NO SON INDEPENDIENTES.

Ver cálculos debajo:

In [ ]:
prob_A_intersec_B

0.26808859721082856

In [ ]:
prob_A*prob_B


0.1945682289104492